src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureAppConfigurationProvider.cs

```diff
@@ -1427,33 +1447,36 @@ private void UpdateClientBackoffStatus(Uri endpoint, bool successful)
             _configClientBackoffs[endpoint] = clientBackoffStatus;
         }
 
-        private async Task<bool> HaveCollectionsChanged(
+        private async Task<bool> HavePageChange(
             IEnumerable<KeyValueSelector> selectors,
-            Dictionary<KeyValueSelector, IEnumerable<MatchConditions>> pageEtags,
+            Dictionary<KeyValueSelector, IEnumerable<PageWatcher>> pageWatchers,
             ConfigurationClient client,
             CancellationToken cancellationToken)
         {
-            bool haveCollectionsChanged = false;
+            bool havePageChanged = false;
 
             foreach (KeyValueSelector selector in selectors)
             {
-                if (pageEtags.TryGetValue(selector, out IEnumerable<MatchConditions> matchConditions))
+                Debug.Assert(pageWatchers.ContainsKey(selector));
+
+                if (pageWatchers.TryGetValue(selector, out IEnumerable<PageWatcher> watchers))
```

Comment:
This sequence doesn't make sense. If there is a guarantee that pageWatchers contains selector, then we shouldn't use TryGetValue afterward.

If there isn't a guarantee of that, then we shouldn't be doing a Debug.Assert.